app/views/community/index.html.slim

```diff
@@ -31,8 +31,7 @@ table.users.table.table-striped.table-bordered.table-condensed.table-hover
         td.image = avatar_url(user, size: 40)
         td = link_to user.name_or_handle, user
         td = user.interested_in.map { |key| User::INTERESTS[key].presence }.compact.join(', ')
-        - if signed_in?
-          td = mail_to user.email unless user.hide_email?
+        td = mail_to user.email if can?(:read, user.email)
```

Comment:
I think there's a bug here: this keeps the `td` even if the `can?` method returns false, which breaks the layout a bit. (Try it with a logged out or unconfirmed user.)

I think you need to change this back to be a 2-line conditional.

Reply:
Also, as pointed out below, I think the check should be `can?(:read_email, user)`.

Reply:
Oiii, thanks for catching this. Turns out I was solving the wrong problem. Fixing the ability rules also fixed the layout _and_ what I was trying to do with the new ability rule.